### ISSUE TITLE
Regression in Pyright 1.1.378 for multiple `TypeVar` definitions in one assignment.

### ISSUE THREAD

**Issue by BarrensZeppelin** [microsoft/pyright#8904]
This code:

```python
from typing import TypeVar

K, V = TypeVar("K"), TypeVar("T")

def f(d: dict[K, V]) -> tuple[K, V]:
    ...
```

[Type checks successfully in Pyright 1.1.378](https://pyright-play.net/?pyrightVersion=1.1.378&strict=true&enableExperimentalFeatures=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAqiApgGoCGIAUFQNIA0UpUAvISRSABQBEtPASkZEEZSrwKCaAE2LAowLtIBcUaUgDGMANoMmAXQFQAtAD5YAVwQAbYrsal9yqlFdQAdJ6A).

[Does not type check in Pyright 1.1.379](https://pyright-play.net/?pyrightVersion=1.1.379&strict=true&enableExperimentalFeatures=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAqiApgGoCGIAUFQNIA0UpUAvISRSABQBEtPASkZEEZSrwKCaAE2LAowLtIBcUaUgDGMANoMmAXQFQAtAD5YAVwQAbYrsal9yqlFdQAdJ6A) with
```
Variable not allowed in type expression  (reportInvalidTypeForm)
Variable not allowed in type expression  (reportInvalidTypeForm)
Variable not allowed in type expression  (reportInvalidTypeForm)
Variable not allowed in type expression  (reportInvalidTypeForm)
Type of parameter "d" is partially unknown
  Parameter type is "dict[Unknown, Unknown]"  (reportUnknownParameterType)
Return type, "tuple[Unknown, Unknown]", is partially unknown  (reportUnknownParameterType)
```

**Comment by erictraut** [comment]
This construct is not allowed in the typing spec. Each TypeVar needs to be defined in its own statement to be considered legal for type checkers. Mypy doesn't accept this either. The fact that it worked in previous versions was a bug.

**Comment by erictraut** [comment]
BTW, if you're using Python 3.12, you might want to switch to the new syntax for generics. See [PEP 695](https://peps.python.org/pep-0695/) for details. With the new syntax, you don't need to explicitly construct type variables.

```python
def f[K, V](d: dict[K, V]) -> tuple[K, V]:
    ...
```

**Comment by BarrensZeppelin** [comment]
Right, thanks for clarifying.

I target https://github.com/pypy/pypy which only supports Python 3.10 at the moment. :slightly_smiling_face:
